genson/src/main/java/com/owlike/genson/stream/ValueType.java

```diff
@@ -12,12 +12,16 @@
   BOOLEAN(Boolean.class),
   NULL(null);
 
-  private final Class<?> clazz;
+  private Class<?> clazz;
 
   ValueType(Class<?> clazz) {
     this.clazz = clazz;
   }
 
+  public void setDefaultClass(Class<?> clazz) {
+    this.clazz = clazz;
```

Comment:
Please no... :/ I think an enum should always be immutable.

Reply:
I hear you, but in that case I would argue that default mappings shouldn't be in an enum...

We really need this: we have our own `JsonObject` class that in our case must be used for deserialization of untyped JSON objects instead of plain `Map`. We have also discovered that we need different mapping for int values, and possibly even for doubles.

The defaults are fine when they work, but in our case they don't. Allowing mutation of the mappings within an enum was the easiest way to do it without breaking anything, but if it would make you feel better :-), I can introduce separate mapping layer and modify `UntypedConverter` to use it?

Reply:
Yes doing the mapping UntypedConverter (or in some new converter) sounds fine to me. The mapping would then be based on the enum and not the class in the enum.